vm/systemSmartContracts/validator.go

```diff
@@ -1176,13 +1183,19 @@ func (v *validatorSC) unStakeTokens(args *vmcommon.ContractCallInput) vmcommon.R
 	return vmcommon.Ok
 }
 
-func (v *validatorSC) unStakeValue(registrationData *ValidatorDataV2, unStakeValue *big.Int) vmcommon.ReturnCode {
-	registrationData.TotalStakeValue.Sub(registrationData.TotalStakeValue, unStakeValue)
-	if registrationData.TotalStakeValue.Cmp(zero) < 0 {
-		v.eei.AddReturnMessage("contract error on unStakeTokens function, total stake < unstake value")
+func (v *validatorSC) processUnStakeValue(registrationData *ValidatorDataV2, unStakeValue *big.Int) vmcommon.ReturnCode {
+	unstakeValueIsOk := unStakeValue.Cmp(v.minUnstakeTokensValue) >= 0 || unStakeValue.Cmp(registrationData.TotalStakeValue) == 0
+	if !unstakeValueIsOk {
+		v.eei.AddReturnMessage("can not unstake the provided value either because is under the minimum threshold or " +
+			"is not the value left to be unStaked")
+		return vmcommon.UserError
+	}
+	if unStakeValue.Cmp(registrationData.TotalStakeValue) > 0 {
+		v.eei.AddReturnMessage("can not unstake a bigger value than the possible allowed value which is " + registrationData.TotalStakeValue.String())
 		return vmcommon.UserError
 	}
 
+	registrationData.TotalStakeValue.Sub(registrationData.TotalStakeValue, unStakeValue)
```

Comment:
also here no backwardscompatibility issue?

Reply:
this is ok. flag check is before.

Reply:
ok